src/main/java/com/mishep/web/servlets/GetCalcServlet.java

```diff
@@ -0,0 +1,88 @@
+package com.mishep.web.servlets;
+
+import com.mishep.web.utils.GetCalculate;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.ArrayDeque;
+import java.util.StringTokenizer;
+
+@WebServlet(
+        name = "GetCalcServlet",
+        urlPatterns = {"/calc/result"}
+)
+public class GetCalcServlet extends HttpServlet {
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) {
+        try {
+            String result = "";
+            String tokenStr = GetCalculate.getReversePolishNotation(((String) req.getSession().getAttribute("equation")).replaceAll("\\s", ""));
+            try {
+                result = getResult(req, tokenStr);
+            } catch (IllegalArgumentException e) {
+                resp.setStatus(409);
+            } finally {
+                if (resp.getStatus() != 409) {
+                    resp.setStatus(200);
+                    (resp.getWriter()).write(result);
+                }
+            }
+            resp.getWriter().flush();
+            resp.getWriter().close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private String getResult(HttpServletRequest req, String tokenStr) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The method 'getResult(HttpServletRequest, String)' has an NPath complexity of 5461](https://app.codacy.com/app/thejerome/IFMO_Servlets_20190915/pullRequest?prid=4589206)